src/app/workspace/resource/resource-instance-form/select-properties/switch-properties/switch-properties.component.ts

```diff
@@ -20,23 +20,20 @@ export class SwitchPropertiesComponent implements OnInit, AfterViewInit {
 
     @Input() formName: string;
 
+    @Input() isRequiredProp: boolean;
+
     mode = 'create';
     constants = Constants;
 
     constructor() { }
 
     ngOnInit(): void {
-        // console.log('prop', this.property);
-    }
-
-    ngAfterViewInit() {
-        // console.log('createValueComponent', this.createValueComponent);
-        // this.saveNewValue();
+        // convert from boolean (1/0) to boolean (true/false)
+        this.isRequiredProp = !!+this.isRequiredProp;
```

Comment:
Maybe you can explain why this is necessary (Input provided by KeyValuePair is stored as a number)

Reply:
done in fb3a064